eng-wasm/src/crypto_wasm.rs

```diff
@@ -4,54 +4,30 @@ use eng_pwasm_abi::types::U256;
 use rand_wasm::Rand;
 
 const SYMMETRIC_KEY_SIZE: usize = 32;
-const IV_SIZE: usize = 96 / 8;
-const PAYLOAD_SIZE: usize = 1024;
 
-pub type IV = [u8; IV_SIZE];
 pub type SymmetricKey = [u8; SYMMETRIC_KEY_SIZE];
 
+fn extra_size_for_cypher() -> usize {
+    unsafe {(external::get_iv_size() + external:: get_tag_size()) as usize }
+}
+
+
 pub fn generate_key() -> SymmetricKey {
     let key_int: U256 = Rand::gen();
     let key = H256::from(key_int);
     key.0
 }
 
 pub fn encrypt(message: &[u8], key: &SymmetricKey) -> Vec<u8> {
```

Comment:
Should add a little bit of text, or at least a reference as to why this works the way it does

Reply:
Added comments at the beginning of the file that AES-256-GCM is used. One has to be familiar with the algorithm to understand.